### ISSUE TITLE
v_out = torch.cat((v_out_h + self.rel_h, v_out_w + self.rel_w), dim=1)

### ISSUE THREAD

**Issue by scutzck033** [leaderj1001/Stand-Alone-Self-Attention#7]
Hi, are the following codes wrong?
"v_out_h, v_out_w = v_out.split(self.out_channels // 2, dim=1)
v_out = torch.cat((v_out_h + self.rel_h, v_out_w + self.rel_w), dim=1)"
shouldn't it be something as follows?
"k_out_h, k_out_w = k_out.split(self.out_channels // 2, dim=1)
k_out = torch.cat((k_out_h + self.rel_h, k_out_w + self.rel_w), dim=1)"
Because the relative distance embedding is supposed to added to the "key" instead of the "value", right?

**Comment by leaderj1001** [comment]
Thanks for your comments.
I'll check it and training it again.
Thank you :)

**Comment by yassouali** [comment]
I think you're right, the relative embedding are added to the keys and not the values. As detailed in eq. 3 of the paper. But instead of changing the values of `k_out`. A new variable needs to be created.

I think it should be something like:

```python
k_out_h, k_out_w = k_out.split(self.out_channels // 2, dim=1)
k_out_rel = torch.cat((k_out_h + self.rel_h, k_out_w + self.rel_w), dim=1)

# And then
out = q_out * k_out + q_out * k_out_rel
```
